bio/agat/wrapper.py

```
@@ -0,0 +1,505 @@
+def get_gff_basename(snake_input):
+    """
+    Many Agat commands base their output file name on GTF/GFF input file
+    name. We set-up the GFF/GTF base name extraction here once for all.
+
+    Search the GFF/GTF input file name among the list of possible
+    """
+    # One of the output file is build on the GFF/GTF input file
+    gff_basename = ""
+    gff_keys = ("f", "ref", "reffile", "gff", "gff3", "gtf")
+    for key in gff_keys:
+        gff = snakemake.input.get(key)
```

Ruff (0.11.9)

Undefined name `snakemake`

(F821)